### PR TITLE
Autreg 58 openapi lambdab

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -59,7 +59,9 @@ dependencies {
     compile group: 'com.amazonaws', name: 'aws-java-sdk-secretsmanager', version: '1.11.+'
     compile group: 'com.amazonaws', name: 'aws-java-sdk-iam', version: '1.11.+'
     compile group: 'com.amazonaws', name: 'aws-java-sdk-api-gateway', version: '1.11.+'
-
+   
+   
+    compile group: 'com.amazonaws', name: 'aws-java-sdk-codepipeline', version: '1.11.442'
```
Comment (janbj) [inline build.gradle]: Do we need more structure to versioning soon?

File: invoke.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+
+sam local invoke  Init -e apiUpdate.json  -d 5005
```
Comment (janbj) [inline invoke.sh]: More descriptive filename?

File: src/main/java/no/bibsys/Application.java [hunk 3]
```diff
@@ -78,13 +73,7 @@ public void createStacks() throws IOException {
         stackBuilder.createStacks();
     }
 
-    public void updateLambdaTrustRole() {
```
Comment (janbj) [inline src/main/java/no/bibsys/Application.java]: Nice!

File: src/main/java/no/bibsys/apigateway/ApiExporter.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            GetExportRequest request = new GetExportRequest().withRestApiId(api.getId())
+                .withStageName(stage).withExportType("oas30").withParameters(requestParameters);
+            GetExportResult result = apiGateway
+                .getExport(request);
+            String swaggerFile = new String(result.getBody().array());
+            return swaggerFile;
+
+        }
+
+        return null;
```
Comment (janbj) [inline src/main/java/no/bibsys/apigateway/ApiExporter.java]: Use Optional?

File: src/main/java/no/bibsys/apigateway/ApiExporter.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            .filter(api -> api.getName().contains(config.getProjectId()))
+            .filter(api -> api.getName().contains(config.getNormalizedBranchName()))
+            .filter(api -> api.getName().contains(stage))
+            .findFirst();
+
+        if (apiOpt.isPresent()) {
+            RestApi api = apiOpt.get();
+
+            GetExportRequest request = new GetExportRequest().withRestApiId(api.getId())
+                .withStageName(stage).withExportType("oas30").withParameters(requestParameters);
```
Comment (janbj) [inline src/main/java/no/bibsys/apigateway/ApiExporter.java]: Constant for visibility?

File: src/main/java/no/bibsys/handler/ApiGatewayHandlerTemplate.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    }
+
+
+    @Override
+    protected void writeOutput(I input,O output) throws IOException {
+        String outputString = objectMapper.writeValueAsString(output);
+        GatewayResponse gatewayResponse = new GatewayResponse(outputString);
+        String responseJson = objectMapper.writeValueAsString(gatewayResponse);
+        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(outputStream));
```
Comment (janbj) [inline src/main/java/no/bibsys/handler/ApiGatewayHandlerTemplate.java]: Use try-with-resources?

File: src/main/java/no/bibsys/handler/CodePipelineFunctionHandler.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    private final transient AWSCodePipeline pipeline=AWSCodePipelineClientBuilder.defaultClient();
+
+    public CodePipelineFunctionHandler() {
+        super(CodePipelineEvent.class);
+    }
+
+    @Override
+    protected final CodePipelineEvent parseInput(InputStream inputStream) throws IOException {
+        String jsonSting= IoUtils.streamToString(inputStream);
+        System.out.println(jsonSting);
```
Comment (janbj) [inline src/main/java/no/bibsys/handler/CodePipelineFunctionHandler.java]: We should introduce proper logging to CloudWatch over print statements

File: src/main/java/no/bibsys/handler/CodePipelineFunctionHandler.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+            .withType(FailureType.JobFailed);
+        PutJobFailureResultRequest failure=new PutJobFailureResultRequest()
+            .withJobId(input.getId()).withFailureDetails(failureDetails);
+        pipeline.putJobFailureResult(failure);
+        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(outputStream));
+        writer.write(outputString);
+        writer.close();
+    }
+
+//    private String createContinuationToken(CodePipelineEvent input) throws JsonProcessingException {
```
Comment (janbj) [inline src/main/java/no/bibsys/handler/CodePipelineFunctionHandler.java]: Remove commented code

File: src/main/java/no/bibsys/handler/requests/CodePipelineEvent.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    public CodePipelineEvent(String id) {
+        this.id = id;
+    }
+
+
+    public static CodePipelineEvent create(String eventJsonString) throws IOException {
+        ObjectMapper mapper = JsonUtils.newJsonParser();
+        JsonNode root = mapper.readTree(eventJsonString);
+        String id= root.get("CodePipeline.job").get("id").asText();
+//        String id=element.get("id").asText();
```
Comment (janbj) [inline src/main/java/no/bibsys/handler/requests/CodePipelineEvent.java]: Why is this commented out?

File: template.yml [hunk 10]
```diff
@@ -34,11 +38,11 @@ Resources:
             post:
               responses:
                 default:
-                  description: A simple string response
+                  description: default response
                   content:
-                    text/plain:
+                    stromg:
```
Comment (janbj) [inline template.yml]: stromg? string?

File: src/main/java/no/bibsys/handler/CodePipelineFunctionHandler.java (hunk 7 above)
Comment (axthosarouris) [reply src/main/java/no/bibsys/handler/CodePipelineFunctionHandler.java]: I agree. I tried to use logging but I was doing something wrong and I could not get logs. 
We should create a task for logging

**Pull request merged**